server/src/pages/shot/text-tool.js

```diff
@@ -0,0 +1,218 @@
+const { ColorPicker } = require("./color-picker");
+
+// Text input height and padding is determined by text size
+// See frame.scss mixin type-box($size)
+// TEXT_INPUT_PADDING_TOP is top padding and
+// TEXT_INPUT_PADDING_LEFT is multiplier used to calculate left padding
+const TEXT_INPUT_PADDING_TOP = 4;
+const TEXT_INPUT_PADDING_LEFT = 0.4;
+const TEXT_INPUT_HEIGHT = 1.3;
+
+// Padding used by draggable div
+const DRAG_PADDING = 10;
+const FONT_STYLE = "sans-serif";
+const FONT_WEIGHT = 900;
+const INIT_FONT_SIZE = 36;
+
+exports.TextTool = class TextTool extends React.Component {
+  constructor(props) {
+    super(props);
+    this.canvas = React.createRef();
+    this.textInput = React.createRef();
+
+    this.canvasCssWidth = props.canvasCssWidth;
+    this.canvasCssHeight = props.canvasCssHeight;
+
+    const INIT_LEFT = (this.canvasCssWidth / 2) - INIT_FONT_SIZE * TEXT_INPUT_PADDING_LEFT;
+    const INIT_TOP = (this.canvasCssHeight / 2) - ((INIT_FONT_SIZE * TEXT_INPUT_HEIGHT) / 2);
+
+    this.state = {
+      left: INIT_LEFT,
+      top: INIT_TOP,
+      textContent: "",
+      color: this.props.color,
+      colorName: this.props.colorName,
+      textSize: "large-text"
+    };
+  }
+
+  componentDidMount() {
+    // Initalize drawing context when user first clicks on canvas
+    this.drawingContext = this.canvas.current.getContext("2d");
+    this.drawingContext.scale(this.props.canvasPixelRatio, this.props.canvasPixelRatio);
+    this.drawingContext.textBaseline = "top";
+
+    this.textInput.current.focus();
+
+    if (this.props.toolbarOverrideCallback) {
+      this.props.toolbarOverrideCallback();
+    }
+  }
+
+  componentDidUpdate(oldProps, oldState) {
+    if (oldState.textSize !== this.state.textSize) {
+       this.props.toolbarOverrideCallback();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.toolbarOverrideCallback) {
+      this.props.toolbarOverrideCallback();
+    }
+  }
+
+  render() {
+    const textToolBox = this.renderTextBox();
+    return <div
+      className={`text-tool-container centered`}
+      onDragOver={this.onDragOver.bind(this)}
+      onDrop={this.onDrop.bind(this)}
+      style={{height: this.props.baseCanvas.style.height, width: this.props.baseCanvas.style.width}}>
+      {textToolBox}
+    </div>;
+  }
+
+  renderTextBox() {
+    const dragDivTopPx = `${this.state.top - DRAG_PADDING}px`;
+    const dragDivLeftPx = `${this.state.left - DRAG_PADDING}px`;
+    const dragDivPadding = `${DRAG_PADDING}px`;
+
+    const dragDivStyles = {
+      position: "absolute",
+      top: dragDivTopPx,
+      left: dragDivLeftPx,
+      padding: dragDivPadding,
+      cursor: "move"
+    };
+
+    return [
+      <canvas ref={this.canvas} id="text-canvas" key="base" width={this.props.baseCanvas.width}
+        height={this.props.baseCanvas.height}
+        style={{width: this.canvasCssWidth,
+                height: this.canvasCssHeight}}>
+      </canvas>,
```

Comment:
We don't need this canvas in the DOM since we aren't really using to display anything.  The canvas can be created on confirmation.

Reply:
It helped to debug by looking at canvas in DOM, but I agree we don't need it , moved it to create inside confirmClick